### PR TITLE
GOBBLIN-1390: Add an option to run a subset of jobs from job config d…

### PR DESCRIPTION
**Pull request by sv2000**
…ir in Gobblin cluster

Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1390

### Description
- [x] Here are some details about my PR, including screenshots (if applicable):
This PR provides an option to run a subset of jobs from the job config directory. This option is specified via a regular expression. 

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Added unit test in JobConfigurationManagerTest.

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by sv2000** [comment]
@autumnust @ZihanLi58 Please review.

File: gobblin-cluster/src/main/java/org/apache/gobblin/cluster/JobConfigurationManager.java [hunk 1]
```diff
@@ -21,12 +21,18 @@
...
 import java.util.List;
 import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
 
 import org.apache.gobblin.cluster.event.CancelJobConfigArrivalEvent;
 import org.apache.gobblin.runtime.job_spec.JobSpecResolver;
+
+import org.apache.parquet.Strings;
```
Comment (autumnust) [inline gobblin-cluster/src/main/java/org/apache/gobblin/cluster/JobConfigurationManager.java:31]: Seems wrong imports

File: gobblin-cluster/src/main/java/org/apache/gobblin/cluster/JobConfigurationManager.java [hunk 2]
```diff
@@ -106,14 +120,31 @@ protected void startUp() throws Exception {
         List<Properties> jobConfigs = SchedulerUtils.loadGenericJobConfigs(properties, this.jobSpecResolver);
         LOGGER.info("Loaded " + jobConfigs.size() + " job configuration(s)");
         for (Properties config : jobConfigs) {
-          postNewJobConfigArrival(config.getProperty(ConfigurationKeys.JOB_NAME_KEY), config);
+          if (!jobsToRun.isPresent() || shouldRun(jobsToRun.get(), config)) {
+            postNewJobConfigArrival(config.getProperty(ConfigurationKeys.JOB_NAME_KEY), config);
+          } else {
+            LOGGER.error("Job {} has been filtered and will not be run in the cluster.", config.getProperty(ConfigurationKeys.JOB_NAME_KEY));
```
Comment (autumnust) [inline gobblin-cluster/src/main/java/org/apache/gobblin/cluster/JobConfigurationManager.java:126]: warn instead?

File: gobblin-cluster/src/main/java/org/apache/gobblin/cluster/JobConfigurationManager.java [hunk 3]
```diff
@@ -106,14 +120,31 @@ protected void startUp() throws Exception {
...
   }
 
+  @VisibleForTesting
+  /**
+   * A helper method to determine if a given job should be submitted to cluster for execution based on the
+   * regex defining the jobs to run.
+   */
+  protected static boolean shouldRun(Pattern jobsToRun, Properties jobConfig) {
+    Matcher matcher = jobsToRun.matcher(jobConfig.getProperty(ConfigurationKeys.JOB_NAME_KEY));
+    if (matcher.matches()) {
```
Comment (autumnust) [inline gobblin-cluster/src/main/java/org/apache/gobblin/cluster/JobConfigurationManager.java:142]: return `matcher.matches()` ?

**Review by autumnust — CHANGES_REQUESTED** [review]

**Review by autumnust — APPROVED** [review]

**Pull request closed without merging**